### ISSUE TITLE
Stops displaying notifications on MM2 after a day or two

### ISSUE THREAD

**Issue by dwburger** [basknol/MMM-PushBulletNotifications#18]
I’m using the Pushbullet module and all is well when I first start the MM2. However, after some time (usually a day or two) messages to Pushbullet no longer display on the MM2. Those messages do appear on other devices connected to Pushbullet. I’ve thought it was because of a wifi drop, but I can remote into the Rpi using PuTTY while the Pushbullet messages don’t display so I don’t think it’s that. Can you suggest where else I should look to solve this problem??? Thanks!

**Comment by basknol** [comment]
Hi @dwburger, 

Did you see any errors in de logging? Stream errors should be logged.
Are you using pm2? Try running the command `pm2 logs mm`

My MM is not on during the night, maybe thats why I never encountered this problem. Because it is reconnecting every day.

**Comment by dwburger** [comment]
Yes, I see these errors, but I don't know what they mean.  

0|mm       | [Error]10:18:05.942 - Error fetching devices. RequestError: Error: self signed certificate in certificate chain
0|mm       | [Error]10:18:06.500 - Push stream error: Error: self signed certificate in certificate chain

I like the idea of turning my MM2 off at night.  Can this process be automated to turn off  and then back on in the morning?  Could you point me to information that will help me get that done?  Thanks!

**Comment by basknol** [comment]
You only get this message after a couple days, not on initial load of your MM?

Here is an explanation of shutting down your MM (Raspberry Pi) based on a schedule using crontab: [how-to-boot-and-shut-down-your-raspberry-pi-safely-and-more-effectively](https://www.lifehacker.co.uk/2015/11/20/how-to-boot-and-shut-down-your-raspberry-pi-safely-and-more-effectively)

Use this command to set a schedule using crontab
`sudo crontab –e`

When the editor appears, add the following line to the bottom of the document to shut down your Pi at 22:45 (10.45pm) every day:

`45 22 * * * sudo shutdown –h`

I'm also using a time switch to switch off the power 5 minutes after the RPi has shut down. This same time switch is switched on in the morning, so the RPi gets power and is booting.

**Comment by dwburger** [comment]
I'll check to see when I get the message(s).  In the meantime, I will check-out the other information you provided.  Thanks!

**Comment by dwburger** [comment]
It's only been a few days, but my MM2 PushBullet module has been working fine since I turned off the 5 GHz radio frequency on my router and forced my RPi to connect to the 2.4GHz WiFi frequency.  I've read online accounts that seem to indicate that the RPi can be less stable at 5 GHz and it seems to be making a difference in my case.  I'll report back if this stability doesn't continue.  I've looked around online to see if I can force the RPi to connect to the 2.4GHz frequency without success.  I'd rather have both frequencies running, but can't yet since I don't know how to force the RPi to connect only at 2.4 GHz.  If anyone reading this knows this can be done, I would appreciate a pointer to the instructions.  Thanks!

**Comment by basknol** [comment]
@dwburger good to hear the module has been working for a few days.

I cannot see a direct relation between the earlier error you reported (self signed certificate in certificate chain) and the WiFi connection/frequency.

I searched on the web for PushBullet in combination with the error message, but did not find anything similar.

Are you using a self signed certificate on your rpi?

**Comment by basknol** [comment]
@dwburger  The MMM-PushBulletNotifications module uses a websocket connection to use the Pushbullet stream. On this page you can test the websocket and streaming connection [https://stream.pushbullet.com/](https://stream.pushbullet.com/)

**Comment by dwburger** [comment]
Honestly, I don't know what any of this means so I don't think I can answer your question.  I'd not heard of "self-signed certificates" before seeing them listed in the error messages.  All I knew at the outset was that the PushBullet module wasn't working a day or two after starting my MM2 and I reported those errors based on your guidance (much appreciated, BTW).  It was a "shot in the dark" to explore whether the 2.4GHz and 5 GHz connections would make a difference.  So far, it has.  I did, however, test the PushBullet stream on my Windows 10 desktop (Chrome) and it connects in a very short period of time (<0.3 sec), repeatedly.
